### ISSUE TITLE
Clicking tags/labels in structure leads to program freeze

### ISSUE THREAD

**Issue by Biorix** [texstudio-org/texstudio#3387]
Often, when I click on the labels on the structure sidebar to navigate
I didn't find consistency on when it does it, but it's pretty often since the last update

I click on the name and the program freezes indefinitely. I need to force close it.

It tried what was proposed on [Issue 1063](https://github.com/texstudio-org/texstudio/issues/1063) but it's not applicable.
~~What might have an impact, is the fact that the project is in a One Drive tree:~~
~~`C:\Users\xxx\OneDrive - company\Documents\Projets\2023\xxx - MàJ\03_Livrables\Rapport\arc-flash`~~

This happens on any projects.

### Environment
- TeXstudio 4.6.3 (git 4.6.3)
- Qt version 6.5.2, compiled with Qt 6.5.2 R
- OS: Windows 11
- TeX distribution: MikTek

### Expected behavior

Click on the name should open the corresponding file

### Actual behavior
Program freeze

### How to reproduce
create a document main
create a subfolder with another .tex document in it
open the main.tex and click on any link on the tree a few times

here is a structure that yield this behavior
[test.zip](https://github.com/texstudio-org/texstudio/files/13502502/test.zip)

### edit

Gave more detailed and ruled out the path and OneDrive hypothesis

**Comment by Biorix** [comment]
Sorry, I know it's not a lot of information.

I also just checked, and it's also having the issue on a normal folder

Here is a screenshot of the tree looks like.
![image](https://github.com/texstudio-org/texstudio/assets/14349329/22a6af96-28a0-4dc8-8787-e428a2ab8de6)

the documents is multi-file, yes

The basic tree would be 

```
main
   |_file1.tex
   |_file2.tex
   |_annexes/
       |_file3.tex
       |_file4.tex
```

It's the same problem shown in the other issue

It's inconsistent, so I cannot predict whether it's going to do it or not

**Comment by Biorix** [comment]
@sunderme I added some info on the issue.
The issue appears regardless of the project, file content or anything

**Comment by sunderme** [comment]
I have clicked the link several times on windows and linux, no crash/freeze.
So,
1) have you checked against given file (test.zip) with no other open document ?
2) do you have non-default settings ?

**Comment by Biorix** [comment]
Yes, I tried without anything else open

I'm not sure about the settings since it does not seems linked to compiler or anything

My install is pretty standard, and I just changed some compilation commands

> no crash/freeze.

It doesn't do it every time. And it's not linked to specific files.
You might need to close and open texstudio a few times
And it happens regardless of how you open it (by clicking a tex file or opening texstudio that displays the last open document automatically).

**Comment by sunderme** [comment]
I work with txs daily and I have not seen this.
Virus scanner ?

**Comment by Biorix** [comment]
You mean scan TeXstudio for viruses?

I don't have an antivirus that could interfere

**Comment by Biorix** [comment]
I also checked again where the .exe is, and it's at C:\Program Files\texstudio
Document is at C:\Users\x\Documents\temp\test

**Comment by sunderme** [comment]
The only way forward I can think of, is that you run txs in a debugger and pause the run when it freezes.
See https://github.com/texstudio-org/texstudio/wiki/Compiling#method-1-qt-sdk

**Comment by sunderme** [comment]
not reproducible

**Comment by JuliusMuschaweck** [comment]
I have the same issue. Since about a year. 
I always work with the newest txs version, Windows 10 Pro 64 bit, on a complex project, a book with 300 pages, several \includeonly \includegraphics etc.
Complete txs deinstallation and reinstallation helps for a while, but then the issue comes up again. 
Then I found that just deleting the texstudio folder in C:\Users\x\AppData\Roaming also helps for a while. Nest time, txs will recreate it. That's not a solution though because all the settings are gone. 
I tried to start txs without the texstudio folder in C:\Users\x\AppData\Roaming, apply the most important settings, and save a copy of the current texstudio folder in C:\Users\x\AppData\Roaming, but that doesn't help either.

**Comment by sunderme** [comment]
so what is the offending setting in the ini then ?

**Comment by JuliusMuschaweck** [comment]
> so what is the offending setting in the ini then ?

I don't know. But I don't think there is an offending setting in the ini. To track such a setting down,
I repeated these steps:
* start txs
* see if the bug appears (click into tree on left, see if txs freezes)
* if yes, save ini file as "bad"
* if no, do "something", 
* close, 
* save texstudio.ini as "still good"

with the following steps for "do something":
* create a new simple template "article" document
* save and compile
* set UI interface from German to English
* set editor font Verdana 12pt
* close simple document
* open complex document, open all include files that appear in the tree on the left. At this point, I can continue to work, as long as txs "sees" the complex document in the current txs session

When I then re-open txs, it loads the complex document and all the include files. And when I click in the document tree on the left, it freezes.

I compared the last good and the bad texstudio.ini file line by line. They are identical except for 
* "Files\Recent%20Files", 
* "Files\Last%20Document", 
* "Files\Default%20File%20Filter" (which changes from TeX-Dateien to TeX Files, reflecting the language change)
No lines deleted, no lines added.

The complex document is quite complex. Dozens of packages used, ten \include files, a hundred \includegraphics files, two bib files, hundreds of bíbliography items, table of contents, custom tables of something, an index. My un-educated guess is that txs tries to analyze this complex document, with quite some success, and then chokes somehow, But I don't know. 

I also installed Qt, cloned the repo, to try same while debugging, but couldn't compile "Missing ZLIB_LIBRARY ZLIB:INCLUDE_DIR", Since I'm not familiar with Qt (using Visual Studio, mostly), I stopped here.

**Comment by JuliusMuschaweck** [comment]
Update: I continued to test which files in the AppData\Roaming\texstudio folder might be offending. I found that lastSession.txss2 is a suspect.

* start txs brand new
* load the complex document with all includes by hand, if it's not automatically loaded at startup
* see if I can reproduce the bug.
* close txs

When I delete all files (not the folders) also in all subfolders incl cache, and leave only "texstudio.ini" and "lastSession.txss2", the bug is reproduced: All tex files are loaded at startup, and clicking in the tree freezes txs

When I delete "lastSession.txss2" and leave all other files, txs starts empty (since it knows nothing about the last session). I load the documents by hand, and the bug disappears.
